tests/functional/endpoint/test_endpoint_search.py

```diff
@@ -0,0 +1,139 @@
+        "tlsftp_server": f"tlsftp://{collection_fqdn}:443",
+        "username": username,
+    }
+    # there are so many null and False values in a typical item, it's worth packing them
+    # in strings and filling them in loops -- makes the document above more readable
+    null_keys = """
+        acl_max_expiration_period_mins authentication_assurance_timeout
+        authentication_policy_id contact_email contact_info department description
+        expire_time gcp_connected gcp_paused globus_connect_setup_key host_endpoint
+        host_endpoint_display_name host_endpoint_id host_path https_server info_link
+        keywords last_accessed_time local_user_info_available
+        mapped_collection_display_name mapped_collection_id myproxy_dn oauth_server
+        organization s3_url sharing_target_endpoint sharing_target_root_path
+        subscription_id user_message user_message_link
+    """.split()
+    false_keys = """
+        acl_available acl_editable activated disable_anonymous_writes disable_verify
+        force_encryption force_verify french_english_bilingual high_assurance in_use
+        is_globus_connect mfa_required non_functional requester_pays s3_owner_activated
+    """.split()
+    for k in null_keys:
+        data[k] = None
+    for k in false_keys:
+        data[k] = False
+    return data
+
+
+@pytest.fixture
+def singular_search_response():
+    collection_id = str(uuid.uuid4())
+    endpoint_id = str(uuid.uuid4())
+    return RegisteredResponse(
+        service="transfer",
+        path="/endpoint_search",
+        metadata={
+            "collection_id": collection_id,
+            "endpoint_id": endpoint_id,
+        },
+        json={
+            "DATA": [_make_mapped_collection_search_result(collection_id, endpoint_id)],
+            "DATA_TYPE": "endpoint_list",
+            "has_next_page": False,
+            "limit": 25,
+            "offset": 0,
+        },
+    )
+
+
+def test_search_shows_collection_id(run_line, singular_search_response):
+    singular_search_response.add()
+    meta = singular_search_response.metadata
+    collection_id = meta["collection_id"]
+    endpoint_id = meta["endpoint_id"]
+
+    result = run_line("globus endpoint search mytestquery")
+
+    lines = result.output.rstrip("\n").split("\n")
+    assert len(lines) == 3
+    assert endpoint_id not in lines[-1]
+    assert collection_id in lines[-1]
+
+
+@pytest.mark.parametrize(
+    "entity_type",
+    (
+        "GCP_mapped_collection",
+        "GCP_guest_collection",
+        "GCSv5_endpoint",
+        "GCSv5_mapped_collection",
+        "GCSv5_guest_collection",
+    ),
+)
+def test_search_can_send_entity_type_parameter(
+    run_line, singular_search_response, entity_type
+):
+    singular_search_response.add()
+    run_line(
+        [
+            "globus",
+            "endpoint",
+            "search",
+            "mytestquery",
+            "--filter-entity-type",
+            entity_type,
+        ]
```

Comment:
Now _this_ is where I would prefer to see a `.split()` call! 😀 Like

```python
cmd = f"globus endpoint search mytestquery --filter-entity-type {entity_type}"
run_line(cmd.split())
```

Reply:
Interesting; I disagree.

`run_line(list | str)` lets us use whichever form fits the usage. In this case, we want to make sure that some variable, `entity_type`, is passed as the value for `--filter-entity-type`.

Consider this alternative way of writing what you suggest:
```python
run_line(
    f'globus endpoint search mytestquery --filter-entity-type "{entity_type}"'
)
```

Is it more correct? That depends on the `shlex` usage which happens in `run_line`, and whether or not `entity_type` can contain a space (which we can see above, it does not).

So in this case, no difference.

But I think the right _habit_ is to be passing variable arguments in the list form, which removes any question of `shlex.split` behavior from the tests.

Reply:
Good point!